packages/siopv2-oid4vp-op-auth/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -382,21 +425,40 @@ export class DidAuthSiopOpAuthenticator implements IAgentPlugin {
       context,
     )
 
-    const contentType = response.headers.get('content-type') || ''
+    const contentType = response?.headers.get('content-type') || ''
     let responseBody: any = null
 
-    const text = await response.text()
+    const text = await response?.text()
     if (text) {
       responseBody = contentType.includes('application/json') || text.startsWith('{') ? JSON.parse(text) : text
     }
 
     return {
       body: responseBody,
-      url: response.url,
-      queryParams: decodeUriAsJson(response.url),
+      url: response?.url,
+      queryParams: response && decodeUriAsJson(response?.url),
```

Comment:
Do not allow response to be undefined

Reply:
Already throwing an error from the siopSendAuthorizationResponse(...) method if there are nor PDs neither DCQL query, thus the response is not undefined anymore. Removed the 'response &&' at line 439